src/content/reference/react/cache.md

```diff
@@ -0,0 +1,485 @@
+
+- React will invalidate the cache for all memoized functions for each server request. 
+- Each call to `cache` creates a new function. This means that calling `cache` with the same function multiple times will return different memoized functions that do not share the same cache.
+- `cachedFn` will also cache errors. If `fn` throws an error for certain arguments, it will be cached, and the same error is re-thrown when `cachedFn` is called with those same arguments.
+- `cache` is for use in [Server Components](https://react.dev/blog/2023/03/22/react-labs-what-we-have-been-working-on-march-2023#react-server-components) only. There are plans to introduce `cache` for Client Components, but it is not a supported feature today. There are plans to add a warning when `cache` is used in Client Components.
+
+---
+
+## Usage {/*usage*/}
+
+### Cache an expensive computation {/*cache-expensive-computation*/}
+
+Use `cache` to skip duplicate work.
+
+```js [[1, 7, "getUserMetrics(user)"],[2, 13, "getUserMetrics(user)"]]
+import {cache} from 'react';
+import calculateUserMetrics from 'lib/user';
+
+const getUserMetrics = cache(calculateUserMetrics);
+
+function Profile({user}) {
+  const metrics = getUserMetrics(user);
+  // ...
+}
+
+function TeamReport({users}) {
+  for (let user in users) {
+    const metrics = getUserMetrics(user);
+    // ...
+  }
+  // ...
+}
+```
+
+If the same `user` object is rendered in both `Profile` and `TeamReport`, the two components can share work and only call `calculateUserMetrics` once for that `user`. 
+
+Assume `Profile` is rendered first. It will call <CodeStep step={1}>`getUserMetrics`</CodeStep>, and check if there is a cached result. Since it is the first time `getUserMetrics` is called with that `user`, there will be a cache miss. `getUserMetrics` will then call `calculateUserMetrics` with that `user` and write the result to cache. 
+
+When `TeamReport` renders its list of `users` and reaches the same `user` object, it will call <CodeStep step={2}>`getUserMetrics`</CodeStep> and read the result from cache.
+
+<Pitfall>
+Calling different memoized functions will read from different caches. To access the same cache, components must call the same memoized function.
+
+```js [[1, 7, "getWeekReport = cache(calculateWeekReport)"], [1, 8, "getWeekReport(cityData)"], [2, 17, "getWeekReport = cache(calculateWeekReport)"], [2, 20, "getWeekReport(cityData)"]]
+// Temperature.js
+import {cache} from 'react';
+import {calculateWeekReport} from './report';
+
+export function Temperature({cityData}) {
+  // 🚩 Wrong: Calling `cache` in component creates new `getWeekReport` for each render
+  const getWeekReport = cache(calculateWeekReport);
+	const report = getWeekReport(cityData);
+  // ...
+}
+
+// Precipitation.js
+import {cache} from 'react';
+import {calculateWeekReport} from './report';
+
+// 🚩 Wrong: `getWeekReport` is only accessible for `Precipitation` component.
```

Comment:
couldn't the code for `Precipitation.js` be correct depending on the relationship between the Temperature component and `Precipitation.js`? e.g. if the Precipitation component was the parent component of the Temperature component the code for `Precipitation.js` might be correct as is, as long as it made the `getWeekReport` function created in `Precipitation.js` is made accessible to the Temperature component.

Reply:
Hmm.. in a way, but the way this example is written doesn't demonstrate that use-case. 

Like yes you can re-write this example to be an example that does demonstrate cache-sharing but in its current implementation it doesn't. It is implicit that these components are rendered as siblings, I could make that clearer with an `App` component.